zerver/lib/actions.py

```diff
@@ -3021,11 +3021,17 @@ def do_change_icon_source(realm: Realm, icon_source: str, log: bool=True) -> Non
 def do_change_plan_type(user: UserProfile, plan_type: int) -> None:
     realm = user.realm
     old_value = realm.plan_type
-    realm.plan_type = plan_type
-    realm.save(update_fields=['plan_type'])
-    RealmAuditLog.objects.create(event_type=RealmAuditLog.REALM_PLAN_TYPE_CHANGED,
-                                 realm=realm, acting_user=user, event_time=timezone_now(),
-                                 extra_data={'old_value': old_value, 'new_value': plan_type})
+    with transaction.atomic():
```

Comment:
I'm not sure about this transaction.atomic; we don't seem to use them anywhere else in actions.py. Maybe worth opening a thread on #development-help.

If we do keep it, only the realm.save() and the RealmAuditLog entries need to be in it.